homeassistant/components/evohome.py

```diff
@@ -0,0 +1,606 @@
+"""Support for Honeywell evohome (EMEA/EU-based systems only).
+
+Support for a temperature control system (TCS, controller) with 0+ heating
+zones (e.g. TRVs, relays) and, optionally, a DHW controller.
+
+A minimal configuration.yaml is as as below:
+
+evohome:
+  username: !secret evohome_username
+  password: !secret evohome_password
+
+# This config parameter is presented with its default value:
+# location_idx: 0        # if you have more than 1 location, use this
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/evohome/
+"""
+
+# Glossary
+# TCS - temperature control system (a.k.a. Controller, Master), which can
+# have up to 13 Slaves:
+#   0-12 Heating zones (a.k.a. Zone), and
+#   0-1 DHW controller, (a.k.a. Boiler)
+
+import logging
+from datetime import datetime, timedelta
+import requests
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    ClimateDevice,
+
+    SUPPORT_OPERATION_MODE,
+    SUPPORT_AWAY_MODE,
+)
+
+from homeassistant.const import (
+    CONF_USERNAME,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+
+    EVENT_HOMEASSISTANT_START,
+    PRECISION_TENTHS,
+
+    TEMP_CELSIUS,
+
+    HTTP_BAD_REQUEST,
+    HTTP_TOO_MANY_REQUESTS,
+)
+
+from homeassistant.core import callback
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+
+REQUIREMENTS = ['evohomeclient==0.2.7']
+
+_LOGGER = logging.getLogger(__name__)
+
+# Usually, only the controller does client api I/O during update() to pull
+# current state - the exception is when zones pull their own schedules.
+# However, any entity can call methods that will (eventually) change state.
+PARALLEL_UPDATES = 0
+
+# these are specific to this component
+DOMAIN = 'evohome'
+DATA_EVOHOME = 'data_' + DOMAIN
+DISPATCHER_EVOHOME = 'dispatcher_' + DOMAIN
+
+MIN_TEMP = 5
+MAX_TEMP = 35
+SCAN_INTERVAL_DEFAULT = 180
+SCAN_INTERVAL_MAX = 300
+
+CONF_LOCATION_IDX = 'location_idx'
+
+# Validation of the user's configuration.
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_LOCATION_IDX, default=0): cv.positive_int,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+# these are for the controller's opmode/state and the zone's state
+EVO_RESET = 'AutoWithReset'
+EVO_AUTO = 'Auto'
+EVO_AUTOECO = 'AutoWithEco'
+EVO_AWAY = 'Away'
+EVO_DAYOFF = 'DayOff'
+EVO_CUSTOM = 'Custom'
+EVO_HEATOFF = 'HeatingOff'
+
+TCS_MODES = [
+    EVO_RESET,
+    EVO_AUTO,
+    EVO_AUTOECO,
+    EVO_AWAY,
+    EVO_DAYOFF,
+    EVO_CUSTOM,
+    EVO_HEATOFF
+]
+
+# bit masks for dispatcher packets
+EVO_MASTER = 0x01
+EVO_SLAVE = 0x02
```

Comment:
We should not use master/slave. Use other words like coordinator

Reply:
OK, switched to `PARENT`, `CHILD` (mainly as same length text).